### DISCUSSION TITLE
Question: what does the "subcribe" option on the endpoint.initiate do?

### DISCUSSION THREAD

**Discussion by einarq** [reduxjs/redux-toolkit#3784] [General]
Wasn't able to find any docs on how this is supposed to work. 

I often end up looking up data on the fly, usually in code that hasn't been completely migrated to RTK and RTKQ yet.
It results in code like this:
```

        const subscription = dispatch(
          endpoints.getNode.initiate({
            id,
          })
        );
        const node = await subscription.unwrap();
        subscription.unsubscribe();

```

I'm under the impression that its a good idea to unsubscribe here to make sure I don't re-run this when its not needed. This is only mean to be on the fly.

I discovered the "subscribe" option that we can pass to .initiate(), in the second argument, to make it look like this:
```

        const subscription = dispatch(
          endpoints.getNode.initiate({
            id,
          }, {
            subscribe: false,
          })
        );
        const node = await subscription.unwrap();

```

However, that doesn't work the way I expected it to, as it never gets the data from the query if it isn't already loaded.

**Comment by phryneas** [comment]
Yeah, currently unfortunately `subscribe` doesn't really work unless you already have data in the store - that's also why it's not explicitly documented.

https://github.com/reduxjs/redux-toolkit/pull/3709 would change that.

Meanwhile your current approach is perfectly fine.

**Reply by einarq** [reply]
Current approach is fine, but a bit cumbersome :)

Hoping that gets merged then, thanks.

**Reply by einarq** [reply]
Is this now fixed in RTK 2?

**Reply by phryneas** [reply]
@einarq should be, #3709 is part of 2.0
